### PR TITLE
Agreement map vector

### PR DESCRIPTION
**Pull request by GregoryPetrochenkov-NOAA**
Pull request addressing #117 

## Additions

- homogenize/vectorize.py - Provides the ability to vectorize raster datasets

## Removals

-

## Changes

- accessors/gval_xarray.py - Provide support for tracking vector benchmark data and returning vector agreement maps
- cases_accesors.py - Add cases to test vector compute agreement map procedure
- test_acessors.py - Add testing for vector compute agreemnet map procedure
- cases_homogenize.py - Add cases to test vectorize operation
- test_homogenize.py - Add tests for vectorize operation

### CODE REVIEW

File: src/gval/accessors/gval_xarray.py [hunk 1]
```diff
@@ -32,6 +33,7 @@ class GVALXarray:
     def __init__(self, xarray_obj):
         self._obj = xarray_obj
         self.data_type = type(xarray_obj)
+        self.benchmark_format = "raster"
```
Comment (fernando-aristizabal) [inline src/gval/accessors/gval_xarray.py:36]: Should the benchmark format be automatically detected and agreement map format be designated?

File: src/gval/accessors/gval_xarray.py [hunk 2]
```diff
@@ -135,6 +137,7 @@ def categorical_compare(
                 benchmark_map=benchmark_map,
                 rasterize_attributes=rasterize_attributes,
             )
+            self.benchmark_format = "vector"
```
Comment (fernando-aristizabal) [inline src/gval/accessors/gval_xarray.py:140]: Same here?

File: src/gval/accessors/gval_xarray.py [hunk 3]
```diff
@@ -160,6 +163,9 @@ def categorical_compare(
             encode_nodata=encode_nodata,
         )
 
+        if self.benchmark_format == "vector":
```
Comment (fernando-aristizabal) [inline src/gval/accessors/gval_xarray.py:166]: Maybe change to agreement_map_format?

File: src/gval/accessors/gval_xarray.py (hunk 1 above)
Comment (GregoryPetrochenkov-NOAA) [reply src/gval/accessors/gval_xarray.py]: Changed the name to agreement_map_format

**Pull request merged**